### PR TITLE
RUN-1315: improve prop iteration

### PR DESCRIPTION
**Pull request by Domiii**
* Only use `Runtime.getProperties` CDT command to iterate properties
* https://linear.app/replay/issue/RUN-1315#comment-f8f54931
* Tested [here](https://github.com/replayio/build-test-dashboard/blob/master/7d5b7168-e9ba-4f72-bcbd-0549c9dce532): 11 failed vs. 133 passed ✔
* Tested again [here](https://github.com/replayio/build-test-dashboard/blob/master/6c4075df-b5a2-4712-8f37-91559a3f2914): 8 vs. 92 ✔
* Tested again [here](https://github.com/replayio/build-test-dashboard/blob/master/50cf95b5-2b3a-4401-b512-b72011ed646c): 13 vs. 87 ✔👀

### CODE REVIEW

**Comment by bhackett1024** [comment]
@Domiii this PR is marked as a draft, can you either mark it as ready for review or remove the review requests?

**Comment by Domiii** [comment]
@bhackett1024 
> @Domiii this PR is marked as a draft, can you either mark it as ready for review or remove the review requests?

Noted. I won't send out drafts anymore, and reduce everyone's time spent.

**Comment by jasonLaster** [comment]
Drafts are great. I think Brian's point is that you also marked him as a reviewer.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -145,6 +145,18 @@ function assert(v, msg = "") {
...
 }
 
+/**
+ * Custom prop iterator (RUN-1315).
+ */
+function* iterateProps(o) {
+  // NOTE: symbols cannot be otherwise iterated.
+  yield *Object.getOwnPropertySymbols(o);
+  
+  for (const k in o) {
```
Comment (kannanvijayan) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:155]: For consistency's sake, wouldn't having this second part be:
```
yield *Object.getOwnPropertyNames(o);
```
be more appropriate?

**Review by kannanvijayan — APPROVED** [review]

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc (hunk 1 above)
Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:155]: @kannanvijayan, The original "fix" here was to actually get rid of `getOwnPropertyNames`. However, it turns out that creating the iterator itself in the `for ... in` ends up also creating the set of all property names. The new fix that's coming will forego all of this, and use a modified version of `Runtime.getProperties` exclusively instead.

Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:155]: However, `getOwnPropertyNames` is still way slower than the `for ... in` solution.

**Pull request merged**